components/brave_ads/core/internal/common/url/url_util.cc

```diff
@@ -40,36 +48,52 @@ GURL ReplaceUrlBraveHostWithChromeHost(const GURL& url) {
 }  // namespace
 
 GURL GetUrlWithEmptyQuery(const GURL& url) {
+  CHECK(url.is_valid());
+
   return GURL(base::StrCat(
       {url.scheme(), url::kStandardSchemeSeparator, url.host(), url.path()}));
 }
 
-bool SchemeIsSupported(const GURL& url) {
+bool DoesSupportUrl(const GURL& url) {
+  CHECK(url.is_valid());
+
   if (url.SchemeIs(url::kHttpsScheme)) {
+    // Always support https:// scheme.
     return true;
   }
 
-  // We must replace the brave:// schema with chrome:// due to GURL not parsing
-  // brave:// schemas.
+  // We must replace the brave:// scheme with chrome:// due to GURL not parsing
+  // brave:// schemes.
   const GURL modified_url = ReplaceUrlBraveHostWithChromeHost(url);
-
   if (!modified_url.SchemeIs(kChromeScheme)) {
     return false;
   }
 
   const std::string host_name = modified_url.host();
   if (host_name == kRewardsHostName || host_name == kSyncHostName ||
       host_name == kWalletHostName) {
+    // Support chrome://rewards, chrome://sync and chrome://wallet
     return true;
   }
 
   if (host_name == kSettingsHostName) {
-    if (modified_url.path() == kSearchEnginesPath) {
-      return true;
-    }
-
-    if (modified_url.path() == kSearchPath) {
-      return true;
+    if (modified_url.path() == kSearchEnginesPath ||
+        modified_url.path() == kSearchPath) {
+      if (!modified_url.has_query()) {
+        // Support chrome://settings/searchEngines and
+        // chrome://settings/search
+        return true;
+      }
+
+      const std::vector<std::string> components =
```

Comment:
Since we know the format of the query string for `chrome://settings`, you could also use https://source.chromium.org/chromium/chromium/src/+/refs/heads/main:net/base/url_util.h;l=86;drc=8a22a8785cc10fee3456575fa4497adeef019c7b

Of course, what you have works fine too.